client/container_create.go

```diff
@@ -21,6 +23,11 @@ type configWrapper struct {
 // It can be associated with a name, but it's not mandatory.
 func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, containerName string) (types.ContainerCreateResponse, error) {
 	var response types.ContainerCreateResponse
+
+	if config != nil && config.StopTimeout != nil && versions.LessThan(cli.version, "1.25") {
+		return response, fmt.Errorf("Server only supports API version %s, %q support requires at least %s", cli.version, "stop timeout", "1.25")
+	}
```

Comment:
Could you make this a function, so that all the error messages have a consistent format and wording.

```go
func NewVersionError(version, required, feature string) error {
    ...
}
```

If you make it a method on `Client` you could even drop the `version` argument.

For the wording how about:


"stop timeout" requires API version 1.25, but the Docker server is version 1.24.